daemon/graphdriver/devmapper/deviceset.go

```diff
@@ -2126,9 +2157,11 @@ func (devices *DeviceSet) removeDevice(devname string) error {
 	return err
 }
 
-func (devices *DeviceSet) cancelDeferredRemoval(info *devInfo) error {
+func (devices *DeviceSet) cancelDeferredRemoval(info *devInfo) (bool, error) {
```

Comment:
If you don't like bool, may be we can provide another helper function which only does the actual removal. Rest of the conditions can be checked by caller.

So say provide another funciton __cancelDeferredRmoval(info devInfo), which does not do various check and just calls devicemapper.CancelDeferredRemove(). Now this function will return success if deferred removal as actually cancelled or error.

Caller can check for rest of the conditions and call this only if.
- Deferred removal is enabled.
- There is an active device.
- Device is in deferred remove state.

That way we will get rid of bool return parameter.